[popover] Make element.togglePopover() return a boolean

rwlbuis opened the pull request:
#### 37def53892aecff0087377d61c462eea48b3451f
<pre>
[popover] Make element.togglePopover() return a boolean
<a href="https://bugs.webkit.org/show_bug.cgi?id=257769">https://bugs.webkit.org/show_bug.cgi?id=257769</a>

Reviewed by Tim Nguyen.

Implement togglePopover API change:
<a href="https://github.com/whatwg/html/pull/9393">https://github.com/whatwg/html/pull/9393</a>

* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/togglePopover-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/togglePopover.html: Added.
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::togglePopover):
* Source/WebCore/html/HTMLElement.h:
* Source/WebCore/html/HTMLElement.idl:

Canonical link: <a href="https://commits.webkit.org/265064@main">https://commits.webkit.org/265064@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/053c94da54074610fcf7f69d67ffcda7a7ea118e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9694 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9945 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10190 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11356 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9466 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9705 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11931 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9904 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12391 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9848 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10692 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8251 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/11515 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7996 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8814 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16219 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9094 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8962 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12297 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9454 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/7708 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8636 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/8710 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2327 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12861 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9225 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/html/HTMLElement.cpp:
```diff
@@ -1478,15 +1478,20 @@ ExceptionOr<void> HTMLElement::hidePopover()
+        auto returnValue = hidePopover();
+        if (returnValue.hasException())
+            return returnValue.releaseException();
+    } else if ((!popoverData() || popoverData()->visibilityState() == PopoverVisibilityState::Hidden) && force.value_or(true)) {
```
```suggestion
    } else if (!isPopoverShowing() && force.value_or(true)) {
```

rwlbuis replied:
Nice! Done.

nt1m commented on Source/WebCore/html/HTMLElement.cpp:
```diff
@@ -1478,15 +1478,20 @@ ExceptionOr<void> HTMLElement::hidePopover()
+    }
+    if (!popoverData())
+        return false;
+    return popoverData()->visibilityState() == PopoverVisibilityState::Showing;
```
```suggestion
    return isPopoverShowing();
```

rwlbuis replied:
Done.

rwlbuis committed:
[popover] Make element.togglePopover() return a boolean
https://bugs.webkit.org/show_bug.cgi?id=257769

Reviewed by Tim Nguyen.

Implement togglePopover API change:
https://github.com/whatwg/html/pull/9393

* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/togglePopover-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/popovers/togglePopover.html: Added.
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::togglePopover):
* Source/WebCore/html/HTMLElement.h:
* Source/WebCore/html/HTMLElement.idl:

Canonical link: https://commits.webkit.org/265064@main
